### ISSUE TITLE
Automatic allocation (task-based, etc) continues working for first ten levels after Orb or Rebirth or Reset

### ISSUE THREAD

**Issue by citrusella** [HabitRPG/habitica#7620]
General Info
- UUID: 2d6ef231-50b4-4a22-90e7-45eb97147a2c
- Browser: Firefox 46.0.1
- OS: Windows 7
### Description

I rebirthed with task-based allocation on but intended to turn it off before level 10 when I should have gotten my first batch of attribute points. Then I remembered/found out I couldn't because that section of the stats page is inaccessible before 10. So I waited for level 10, and when I got there I went to turn it off and noticed I had nine points allocated (6 INT, 1 CON, 2 PER) with no unallocated points.This is strange for two reasons:
1. I don't recall it happening at any time (especially not level 10) during the v3 tests when I rebirthed with allocation on there. I know the site isn't entirely equivalent to the v3 tests, but I thought this important to bring up because it was the only other time I'd leveled to ten with allocation on.
2. I'm pretty sure from the past that I should have 10 (or 11?) points to allocate which means one is missing. Theoretically, that would mean it would give me the extra points to allocate next level (I've turned allocation off), but it's strange they weren't already granted...

I'm also a little bummed because I probably would have distributed those differently, but... eh, I can wait until the next time I class cycle during the summer festival.

I said in the Report a Bug guild that (from my inexperienced perspective) it's looks like the auto-allocation kept working after rebirth but didn't grant a point for the first level (for whatever rebirth-related reasons that could have happened). Alys asked me to report here.
#### Console Errors

Nothing.

**Comment by Alys** [comment]
In my opinion, the ideal behaviour would be that after a rebirth, the points for the first 10 levels shouldn't be allocated at all, and I think it should be the same for a new player before they reach level 10 (I'm not sure what the current situation is with that there and I haven't had time to test yet).

If no one objects to that idea in a few days, I'll mark this as help welcome.

@citrusella I've reset your points. You should have 10 unallocated now. Tell us if it's still not right.

**Comment by citrusella** [comment]
> the points for the first 10 levels shouldn't be allocated at all, and I think it should be the same for a new player before they reach level 10

A new player shouldn't currently be able to select an allocation method, so if my previous three times through are any indication, they'll just get the 10 points, unallocated (unless something _else_ is messed up).

And thank you for the points! Everything seems to be working now (I even checked off some tasks I'd just done and leveled and assigned an eleventh point XP).

**Comment by Alys** [comment]
tl;dr the only change needed for this issue is that **both the Orb of Rebirth and Reset** should set `preferences.automaticAllocation` to be `false` regardless of what its previous value was (this is in addition to all the other changes Rebirth and Reset already make, of course).

I did some testing with a new account, following through various settings changes, and below is the current behaviour, in case anyone else is interested. Much of it isn't directly relevant to this issue though.
- a new, level 1 player has 0 points (this is okay)
- when the player reaches level 2, they immediately get two points (I believe there's no special code for this; the level-up code handles the extra point automatically)
- at level 10, they can choose a class and if they do, `flags.classSelected` changes from `false` to `true`
- if they opt out of the class system, `preferences.disableClasses` changes from `false` to `true`
- when they opt back in, `preferences.disableClasses` changes back to `false`
- if they choose automatic allocation, `preferences.automaticAllocation` changes from `false` (actually not specified at all) to `true` (and `preferencecs.allocationMode` contains a string that describes the type of auto allocation)
- when they use the Orb of Rebirth: 
  - **`preferences.automaticAllocation` does NOT change back to `false` but it should**
  - `flags.classSelected` changes to `false`
  - the level changes to 1 and the points to 0 (this is okay)
- when they reach level 2, auto-allocation occurs but for only one point because auto-allocation doesn't correctly handle jumping up more than one level at once (which is basically what happens here from the point of view of attribute points). This is the source of the missing point that @citrusella discovered, however this does NOT need to be worked on in this issue. It will be fixed for the Rebirth/Reset case by the `preferences.automaticAllocation` fix, and it will be fixed for all other cases when https://github.com/HabitRPG/habitrpg/issues/6482 is resolved.
- if they use Reset instead of the Orb of Rebirth, `preferences.automaticAllocation` also does NOT change back to `false` but it should

**Comment by librarianmage** [comment]
I can do this!

**Comment by librarianmage** [comment]
@Alys Question: What is the difference between `reset` and `reroll`?

**Comment by librarianmage** [comment]
@Alys Never mind. They are completely different

**Comment by SabreCat** [comment]
For the curiosity of others who might be following: `reroll` is a legacy name for the Fortify Potion. We could potentially change it in the code for clarity, but it'd be a breaking change for minimal benefit.

**Comment by shervinmathieu** [comment]
If this is not fixed yet I want to pick it up, as well as #7542

**Comment by Alys** [comment]
@velezsarain MathWhiz has a PR for this already, but the other issue isn't being worked on as far as we know. Comment on that one to say you're doing it (no need to ask there; you're welcome to start now).

**Comment by shervinmathieu** [comment]
Oh wow I can't believe I missed the pull request. I must be blind -_-
